src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1419,10 +1391,46 @@ private ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedA
                             newArguments[i] = ((OutDeconstructVarPendingInference)argument).FailInference(this);
                             break;
                         }
+                    case BoundKind.Parameter:
+                    case BoundKind.Local:
+                        {
+                            newArguments[i] = argument.WithWasConverted();
```

Comment:
>newArguments[i] = argument.WithWasConverted(); [](start = 28, length = 46)

It feels like this special handling of locals and parameters belongs in ```BindToTypeForErrorRecovery```. I think we would want all callers of that API to take advantage of it.

Reply:
I want to avoid converting to the corresponding parameter type, and leave the type as-is.  Otherwise a large number of test scenarios are affected.

---
In reply to: [328287208](https://github.com/dotnet/roslyn/pull/38750#discussion_r328287208) [](ancestors = 328287208)